independent-projects/arc/runtime/src/main/java/io/quarkus/arc/Priority.java

```diff
@@ -1,18 +1,25 @@
 package io.quarkus.arc;
 
-import static java.lang.annotation.RetentionPolicy.RUNTIME;
-
 import java.lang.annotation.Retention;
-import java.lang.annotation.Target;
+import java.lang.annotation.RetentionPolicy;
 
 /**
- * This annotation has the same semantics as {@link jakarta.annotation.Priority} except that the {@link Target}
- * meta-annotation is not present. The main motivation is to support method and field declarations, i.e. this annotation
- * can be used for producer methods and fields. Note that this problem is fixed in Common Annotations 2.1.
+ * This annotation has the same semantics as {@link jakarta.annotation.Priority}.
+ * <p>
+ * Prior to Common Annotations 2.1, the {@code jakarta.annotation.Priority} annotation
+ * was meta-annotated {@code @Target({TYPE, PARAMETER})} and so was only usable on class
+ * declarations and method parameters. This annotation was introduced to allow annotating
+ * producer methods and fields.
+ * <p>
+ * Since Common Annotations 2.1, the {@code jakarta.annotation.Priority} is no longer
+ * meta-annotated {@code @Target}, so these two annotations are equivalent.
  * <p>
  * A priority specified by {@link AlternativePriority} and {@link jakarta.annotation.Priority} takes precedence.
+ *
+ * @deprecated use {@link jakarta.annotation.Priority} instead
```

Comment:
We could log a warning/info for this annotation to help users spot it before we get to removal, WDYT?

Reply:
I don't think anyone looks at the warnings we currently produce, and adding more is just an exercise in futility.

What we could do is fail the build if the annotation is used, but that's a pretty harsh measure :-)

Reply:
We have done that before and we can just drop the annotation if we document it properly in the migration guide.

It has been deprecated for a long time and in several Quarkus and RHBQ releases.

Reply:
Fair enough. I don't think this PR is a good vehicle for that, as it affects a few Quarkus core extensions and at least one Quarkiverse extension. I filed #30963 for it.